### DISCUSSION TITLE
Cannot find module '@babel/helper-define-map'

### DISCUSSION THREAD

**Discussion by kevinashworth** [babel/babel#13585] [Q&A]
Briefly, I'm working in a Dockerized environment, not somebody who has paid attention to the details of Babel, and certainly ignorant about how Babel loads helper packages.

This morning I'm getting this error* upon `react-scripts build` in our create-react-app 4.0.3 project, whereas it was working last week. I see a Babel release in the last 24 hours, wondering if it might be part of the cause of my errors? (Because I can't figure out anything that's changed on my end.)

```
Error: foo/src/app.js: Cannot find module '@babel/helper-define-map'
Require stack:
foo/node_modules/babel-preset-react-app/node_modules/@babel/plugin-transform-classes/lib/transformClass.js
foo/node_modules/babel-preset-react-app/node_modules/@babel/plugin-transform-classes/lib/index.js
foo/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/available-plugins.js
foo/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/plugins-compat-data.js
foo/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/normalize-options.js
foo/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/index.js
foo/node_modules/babel-preset-react-app/create.js
foo/node_modules/babel-preset-react-app/index.js
foo/node_modules/react-scripts/node_modules/@babel/core/lib/config/files/plugins.js
foo/node_modules/react-scripts/node_modules/@babel/core/lib/config/files/index.js
foo/node_modules/react-scripts/node_modules/@babel/core/lib/index.js
foo/node_modules/react-scripts/node_modules/babel-loader/lib/index.js
foo/node_modules/loader-runner/lib/loadLoader.js
foo/node_modules/loader-runner/lib/LoaderRunner.js
foo/node_modules/webpack/lib/NormalModule.js
foo/node_modules/webpack/lib/NormalModuleFactory.js
foo/node_modules/webpack/lib/Compiler.js
foo/node_modules/webpack/lib/webpack.js
foo/node_modules/react-scripts/scripts/build.js
```

*slightly edited for privacy, not actually named foo

**Comment by nicolo-ribaudo** [comment]
This has been fixed by https://github.com/babel/babel/pull/12723 in `@babel/plugin-transform-classes@7.12.13`. Please make sure to upgrade to the last version of `@babel/plugin-transform-classes`.

(ref https://github.com/babel/babel/issues/13586)

**Comment by chiptus** [comment]
I see this error. clearning node_modules and reinstalling fixes it for a short time. we don't have a dependency on @babel/plugin-transform-classes

**Reply by chiptus** [reply]
![devDeps](https://user-images.githubusercontent.com/1381655/142383281-9cc78933-4637-4a2f-a866-33b720dd43b5.png)
![deps](https://user-images.githubusercontent.com/1381655/142383320-74c5d30b-f702-489d-9e31-310cf71e9ef4.png)

**Reply by chiptus** [reply]
upgrading @babel/preset-env fixed the issue
